### ISSUE TITLE
Journal Search Improvement

### ISSUE THREAD

**Issue by CelestianGC** [foundryvtt/foundryvtt#9155]
### User Experience

I would like to propose an enhancement to the Foundry VTT system that could improve the user experience when it comes to searching within journals.

Since the introduction of the ability to add "pages" to a journal, users have encountered some challenges when trying to efficiently search through their content. For example, when there is a folder containing multiple journals, each with several pages, it becomes cumbersome to locate specific information using the search function on the Journal tab.

To address this issue, I suggest expanding the search functionality on the Journal tab to search across all journals and sub-pages, looking for matches within the journal sub-page names. If possible, extending the search to include text within the pages themselves would further enhance the user experience.

Currently, the search feature is less effective for users with folders organized by chapters, as it can be difficult to pinpoint the desired information unless they already know its exact location.

In addition to the above suggestion, I recommend adding a search feature to the PDF view. This could display a list of matches in a column, accompanied by a brief context surrounding the matching text, making it easy for users to jump to the relevant section. This functionality would be similar to that found in most PDF viewer applications.

Please note that the introduction of sub-pages within journals has been a fantastic addition to the system. However, refining the search capabilities would significantly improve the overall experience for users navigating their content.

Thank you for your time and consideration. I believe these enhancements would greatly benefit the Foundry VTT community.

**Comment by Mozmonar** [comment]
I would greatly appreciate this as well.

**Comment by kjgonzalez** [comment]
yes please! searching journal content would really push the usefulness of a journal way further. I'd even mention a stretch goal: search across all journals a player / GM can access in a world, if possible

**Comment by SpencerGresoro** [comment]
Probably the single most important feature you can have in a TTRPG VTT environment, especially for a GM. 

PDF search doesn't really cut it, especially without a search summary list.

The ability to search all journal entries, not just the titles, and parse out content quickly, id say, is essential to both players/gms alike. 

The FG search, id say, is the benchmark here.

**Comment by agolp** [comment]
I'd love this as well.

**Comment by Maltavius** [comment]
The Journal Editor also needs Native Search and Replace.
The fact that I can't search and replace a @UUID string that has changed across ALL of my Journals in one go makes me wonder why I would want any documentation inside of Foundry.
